src/sqlfluff/rules/L057.py

```diff
@@ -89,6 +89,22 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                     identifier = identifier[:-1]
                 identifier = identifier.replace(".", "")
 
+            # Spark3 file references for direct file query
+            # are quoted in back ticks to allow for identfiers common
+            # in file paths and regex patterns for path globbing
+            # https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-file.html
+            #
+            # Path Glob Filters (done inline for SQL direct file query)
+            # https://spark.apache.org/docs/latest/sql-data-sources-generic-options.html#path-global-filter  # noqa: E501
+            #
+
+            if (
+                context.dialect.name in ["spark3"]
+                and context.parent_stack
+                and context.parent_stack[-1].name == "FileReferenceSegment"
+            ):
+                return None
```

Comment:
Could we add a test to test/fixtures/rules/std_rule_cases/L057.yml? The other test cases you already added are only tested for parsing, not for rules compliance (as most peoples test SQL cases probably wouldn't pass ALL the rules!)